pkg/util/validation/limits.go

```diff
@@ -38,6 +38,7 @@ const (
 	maxMetadataLengthFlag      = "validation.max-metadata-length"
 	creationGracePeriodFlag    = "validation.create-grace-period"
 	maxQueryLengthFlag         = "store.max-query-length"
+	maxTotalQueryLengthFlag    = "store.max-total-query-length"
```

Comment:
I don't think it should have the `-store` prefix. It has nothing to do with the store. Unfortunately the other `-store.max-query-length`  was inherited from the chunks storage, because at that time the limit was enforced there (see https://github.com/grafana/mimir/issues/2793 too).

Reply:
I would rather prefix it with `query-frontend.`.

Reply:
As described in https://github.com/grafana/mimir/issues/2793, I think we should deprecate `-store.max-query-length` and introduce a new config option for the check done in the querier. Not in this PR obviously, just a longer term plan. I would prefer to not add another CLI flag prefixed with the incorrect `-store.` otherwise it will make even more difficult fixing it in the future.